linux-cachyos-autofdo/PKGBUILD

```diff
@@ -0,0 +1,813 @@
+#
+_processor_opt=${_processor_opt-zen4}
+
+# This does automatically detect your supported CPU and optimizes for it
+_use_auto_optimization=${_use_auto_optimization-}
+
+# Clang LTO mode, only available with the "llvm" compiler - options are "none", "full" or "thin".
+# ATTENTION - one of three predefined values should be selected!
+# "full: uses 1 thread for Linking, slow and uses more memory, theoretically with the highest performance gains."
+# "thin: uses multiple threads, faster and uses less memory, may have a lower runtime performance than Full."
+# "none: disable LTO
+_use_llvm_lto=${_use_llvm_lto-thin}
+
+# Use suffix -lto only when requested by the user
+# y - enable -lto suffix
+# n - disable -lto suffix
+# https://github.com/CachyOS/linux-cachyos/issues/36
+_use_lto_suffix=${_use_lto_suffix-}
+
+# Use suffix -gcc when requested by the user
+# Enabled by default to show the difference between LTO kernels and GCC kernels
+_use_gcc_suffix=${_use_gcc_suffix-y}
+
+# KCFI is a proposed forward-edge control-flow integrity scheme for
+# Clang, which is more suitable for kernel use than the existing CFI
+# scheme used by CONFIG_CFI_CLANG. kCFI doesn't require LTO, doesn't
+# alter function references to point to a jump table, and won't break
+# function address equality.
+_use_kcfi=${_use_kcfi-}
+
+# Build the zfs module in to the kernel
+# WARNING: The ZFS module doesn't build with selected RT sched due to licensing issues.
+# If you use ZFS, refrain from building the RT kernel
+_build_zfs=${_build_zfs-}
+
+# Builds the nvidia module and package it into a own base
+# This does replace the requirement of nvidia-dkms
+_build_nvidia=${_build_nvidia-}
+
+# Builds the open nvidia module and package it into a own base
+# This does replace the requirement of nvidia-open-dkms
+# Use this only if you have Turing+ GPU
+_build_nvidia_open=${_build_nvidia_open-}
+
+
+# Build a debug package with non-stripped vmlinux
+_build_debug=${_build_debug-y}
+
+# Enable AUTOFDO_CLANG for the first compilation to create a kernel, which can be used for profiling
+# This only works for Clang compiled kernels, therefore thin or full lto needs to be set in _use_llvm_lto
+# Workflow:
+# https://cachyos.org/blog/2411-kernel-autofdo/
+# 1. Compile Kernel with _autofdo=y and _build_debug
+# 2. Boot the kernel in QEMU or on your system, see Workload
+# 3. Profile the kernel and convert the profile, see Generating the Profile for AutoFDO
+# 4. Put the profile into the sourcedir and name the profile "perf.afdo" (You can also change the name in _autofdo_profile_name)
+# 5. Enable _autofdo_use
+_autofdo=${_autofdo-y}
+
+# Enable after you have put the profile into the sourcedir
+_autofdo_use=${_autofdo_use-}
+
+# Name for the AutoFDO profile
+_autofdo_profile_name=${_autofdo_profile_name-"perf.afdo"}
+
+
+# ATTENTION: Do not modify after this line
+_is_lto_kernel() {
+    [[ "$_use_llvm_lto" = "thin" || "$_use_llvm_lto" = "full" ]] || [ -n "$_use_kcfi" ]
+    return $?
+}
```

Comment:
We should add a check here if autofdo is used.

Reply:
Yes, can be added.